### DISCUSSION TITLE
Optional with default

### DISCUSSION THREAD

**Discussion by ghost** [cue-lang/cue#1804] [Q&A]
Hi, trying to create shortcuts for optional long labels but fields are being cleared and I can't figure why.

```
#Type: {
    _short?: string
    veryLongLabel?: *_short | string
    ...
}

Alice: #Type & {
    _short: "Alice"
}

// Override works as expected
Bob: #Type & {
  veryLongLabel: "Bob"
}

Charlie: #Type
```

Expected:
```
Alice: {
	veryLongLabel: "Alice"
}
Bob: {
	veryLongLabel: "Bob"
}
Charlie: {}

```

Getting:
```
Alice: {}
Bob: {
	veryLongLabel: "Bob"
}
Charlie: {}
```

Any ideas?

**Update 1:**

@slewiskelly @mvdan Thank's for the responses.

Playing a little bit further with the case left my "typescript sense tingling". In typescript, making a field optional `field?: string` is just syntactic sugar for `field: string | undefined`. If I could do something like that in cue, it should be possible to express optional on the RHS and resolve this with one-liners.

Something like this (https://cuelang.org/play/?id=qCAmdGQzsWw#cue@export@cue):

```
#Type: {
    _short: veryLongLabel & (string | *undefined)
    veryLongLabel: _short & (string | *undefined)
    ...
}

Alice: #Type & {
    _short: "Alice"
}

Bob: #Type & {
  veryLongLabel: "Bob"
}

Charlie: #Type
```
Resulting in
```
Alice: {
	veryLongLabel: "Alice"
}
Bob: {
	veryLongLabel: "Bob"
}
Charlie: {
	// empty, since "veryLongLabel: undefined" would remove the field
}
```

Is there some workaround in cue that would enable something like that?

**Comment by slewiskelly** [comment]
I think you'd need to do something like this:

```cue
#Type: {
    _short?: string

    if _short == _|_ {
        veryLongLabel?: string
    }

    if _short != _|_ {
        veryLongLabel: *_short | string
    }

    ...
}

Alice: #Type & {
    _short: "Alice"
}

// Override works as expected
Bob: #Type & {
  veryLongLabel: "Bob"
}

Charlie: #Type
```

**Reply by mvdan** [reply]
Slight tweak - no need to put `veryLongLabel?: string` behind a conditional, as it's an optional field. https://cuelang.org/play/?id=vbFgJB_ArBt#cue@export@cue
